scripts/build/companion_libs/050-zlib.sh

```diff
@@ -84,6 +84,7 @@ do_zlib_backend() {
     CT_DoLog EXTRA "Configuring zlib"
 
     CT_DoExecLog CFG                                  \
+    CC="${host}-gcc"                                  \
```

Comment:
Argh, found it. GMP tries to outsmart the `AC_CHECK_TOOL` macro: it first does some search for `${CC}` on its own unless the user supplied `CC` environment variable. If the user didn't, and GMP manages to find what it is looking for, it sets CC to whatever it found (`gcc` in our case) and **then** calls `AC_PROG_CC`. Hence, we see the funny message:
````
checking for x86_64-build_pc-linux-gnu-gcc... gcc
````
despite having x86_64-build_pc-linux-gnu-gcc in the path. 

This happens because GMP only prepends the `${host_alias}-` prefix if not cross-compiling. `AC_CHECK_TOOL` always checks for `${host_alias}-tool`  if host triplet is given. As a result, it uses `gcc` but `x86_64-build_pc-linux-gnu-g++`. Funny, huh.

What this means for this PR: you don't need to set CC for all the rest of the configure runs, just for GMP. And, please put a FIXME comment on top of that, something along the lines of:
````
# FIXME GMP ignores --host when searching for CC/GCC if not cross-compiling
# (i.e. if host == build); force it to use the right CC.
````
I'll add fixing the GMP's configure and submitting it upstream to my todo :)